tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs

```diff
@@ -0,0 +1,115 @@
+// <copyright file="SymbolUploadApi.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.Upload
+{
+    internal class SymbolUploadApi : DebuggerUploadApiBase
+    {
+        private const int MaxRetries = 3;
+        private const int StartingSleepDuration = 3;
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<SymbolUploadApi>();
+
+        private readonly IApiRequestFactory _apiRequestFactory;
+        private readonly ArraySegment<byte> _eventMetadata;
+
+        private SymbolUploadApi(
+            IApiRequestFactory apiRequestFactory,
+            IDiscoveryService discoveryService,
+            IGitMetadataTagsProvider gitMetadataTagsProvider,
+            ArraySegment<byte> eventMetadata)
+            : base(apiRequestFactory, gitMetadataTagsProvider)
+        {
+            _apiRequestFactory = apiRequestFactory;
+            _eventMetadata = eventMetadata;
+
+            discoveryService.SubscribeToChanges(c => Endpoint = c.SymbolDbEndpoint);
+        }
+
+        public static IBatchUploadApi Create(
+            IApiRequestFactory apiRequestFactory,
+            IDiscoveryService discoveryService,
+            IGitMetadataTagsProvider gitMetadataTagsProvider,
+            string serviceName)
+        {
+            ArraySegment<byte> GetEventMetadataAsArraySegment()
+            {
+                var sb = new StringBuilder();
+                sb.Append(@"{");
+                sb.Append(@"""ddsource"": ""dd_debugger"",");
+                sb.Append(@$"""service"": ""{serviceName}"",");
+                sb.Append(@$"""runtimeId"": ""{Tracer.RuntimeId}""");
+                sb.Append(@"}");
+                var eventMetadata = sb.ToString();
```

Comment:
It would be more efficient and easier to read to use raw string here I think 🙂 
```suggestion
                var eventMetadata = $$"""
                    {"ddsource":"dd_debugger","service":"{{serviceName}}","runtimeId":"{{Tracer.RuntimeId}}"}
                    """;
```

Reply:
Your suggestion wasn't compiled for me, but I rewrote it slightly different here:
7e1e4a6835f03b3d6dbef617158ce5ecdb406125

Reply:
Weird it didn't compile 🤔 This works in rider:

```csharp
var eventMetadata = $$"""
                      {"ddsource":"dd_debugger","service":"{{serviceName}}","runtimeId":"{{Tracer.RuntimeId}}"}
                      """;
```

but anyway, yours works too! 😄